PMSCore/UserJsonConverter.cs

```diff
@@ -0,0 +1,82 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace PMSCore;
+
+public class UserJsonConverter : JsonConverter<User>
+{
+    public override User? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        if (reader.TokenType != JsonTokenType.StartObject)
+        {
+            return null;
+        }
+
+        reader.Read();
+        if (reader.TokenType != JsonTokenType.PropertyName)
+        {
+            return null;
+        }
+
+        string? propertyName = reader.GetString();
+        if (propertyName != "UserType")
+        {
+            return null;
+        }
+
+        reader.Read();
+        var userType = reader.GetString();
```

Comment:
var userType = reader.GetString(); is missing null handling for user type, maybe this would be better: var userType = reader.GetString() ?? throw new JsonException("UserType is null."); 
what do you think?

Reply:
please check the swich statement bellow which alreadt habdles that case